### PR TITLE
expression, executor: rewrite built-in function uuid.

### PR DESCRIPTION
**Pull request by wusphinx**
uuid refactor

### CODE REVIEW

**Comment by zz-jason** [comment]
plz add type infer test in `plan/typeinfer_test.go`

**Comment by wusphinx** [comment]
@zz-jason already add type infer test in plan/typeinfer_test.go

File: expression/builtin_miscellaneous.go [hunk 1]
```diff
@@ -716,21 +716,23 @@ type uuidFunctionClass struct {
...
-	err := errors.Trace(c.verifyArgs(args))
-	bt := &builtinUUIDSig{newBaseBuiltinFunc(args, ctx)}
-	bt.deterministic = false
-	return bt.setSelf(bt), errors.Trace(err)
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpString)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	bf.tp.Flen = 36
+	sig := &builtinUUIDSig{baseStringBuiltinFunc{bf}}
```
Comment (XuHuaiyu) [inline expression/builtin_miscellaneous.go]: set deterministic to false

File: expression/builtin_miscellaneous_test.go [hunk 2]
```diff
@@ -91,20 +91,21 @@ func (s *testEvaluatorSuite) TestIsIPv4(c *C) {
 
 func (s *testEvaluatorSuite) TestUUID(c *C) {
```
Comment (XuHuaiyu) [inline expression/builtin_miscellaneous_test.go]: add test for isDeterministic.

Comment (wusphinx) [inline expression/builtin_miscellaneous_test.go]: I have done

**Comment by XuHuaiyu** [comment]
LGTM

**Pull request merged**